client.go

```diff
@@ -94,6 +101,26 @@ func NewClient(ctx context.Context, p OpenFGAParams) (*Client, error) {
 				ApiToken: p.Token,
 			},
 		}
+	} else {
+		config.Credentials = &credentials.Credentials{
+			Method: credentials.CredentialsMethodNone,
+		}
+	}
+	if p.HTTPClient != nil {
+		config.HTTPClient = p.HTTPClient
+		// When a custom HTTPClient is provided in OpenFGA configuration,
+		// it does not add authorization headers, so we manually add them here.
+		_, headers := config.Credentials.GetHttpClientAndHeaderOverrides()
+		defaultHeaders := make(map[string]string)
+		if len(headers) != 0 {
+			for idx := range headers {
+				defaultHeaders[headers[idx].Key] = headers[idx].Value
+			}
+		}
+		config.DefaultHeaders = defaultHeaders
+	}
+	if p.Telemetry != nil {
+		config.Telemetry = p.Telemetry
```

Comment:
consider adding an example with the telemetry configured.

Reply:
Added, thank you.